### ISSUE TITLE
--format can push you over max line limit

### ISSUE THREAD

**Issue by liutikas** [shyiko/ktlint#101]
Given this input:
```
package android.arch.persistence.room

import android.arch.persistence.room.processor.Context
import com.google.auto.common.BasicAnnotationProcessor
import com.google.common.collect.SetMultimap
import javax.lang.model.element.Element

class RoomProcessor : BasicAnnotationProcessor() {
    class DatabaseProcessingStep(context: Context) : ContextBoundProcessingStep(context) {
        override fun process(elementsByAnnotation: SetMultimap<Class<out Annotation>, Element>)
                : MutableSet<Element> {
            return mutableSetOf()
        }
    }
}
```

ktlint --format will produce:
```
package android.arch.persistence.room

import android.arch.persistence.room.processor.Context
import com.google.auto.common.BasicAnnotationProcessor
import com.google.common.collect.SetMultimap
import javax.lang.model.element.Element

class RoomProcessor : BasicAnnotationProcessor() {
    class DatabaseProcessingStep(context: Context) : ContextBoundProcessingStep(context) {
        override fun process(elementsByAnnotation: SetMultimap<Class<out Annotation>, Element>): MutableSet<Element> {
            return mutableSetOf()
        }
    }
}
```

Which now will fail line length check of 100 that I have set.

**Comment by shyiko** [comment]
Thank you for the ticket. 
This is related to https://github.com/shyiko/ktlint/issues/96 (`max_line_length` is not yet enforced by the formatter). 

On a separate note, `ktlint --format` should have reported that `max_line_length` is exceeded. This needs to fixed regardless of whether #96 is taken care of or not by the time next release is out.

**Comment by shyiko** [comment]
Closing in favor of #251.
